### PR TITLE
done

### PR DESCRIPTION
**Pull request by Maksymyr**

### CODE REVIEW

**Comment by KatSick** [comment]
Hi,

Thanks for submitting PR!

Can you deploy this app using `github pages`, `surge` or `now` to check functionality ?

File: src/App.js [hunk 1]
```diff
@@ -1,19 +1,161 @@
...
+    else {
+      let sorted = array.sort((item, nextItem) => (item.age < nextItem.age) ? 1 : (item.age > nextItem.age) ? -1 : 0);
+      this.setState({search: sorted, item: sorted[0], checkSortAge: !this.state.checkSortAge})
+    }
+  }
+  handleClear() {
+    let array = this.state.searched.length == 0 ? this.state.data : !this.state.checkSearch ? this.state.searched : this.state.data;
+    let sorted = array.sort((item, nextItem) => (item.id < nextItem.id) ? -1 : (item.id > nextItem.id) ? 1 : 0);
+    this.setState({search: sorted, item: sorted[0]}) 
+      this.setState({
```
Comment (KatSick) [inline src/App.js]: It looks like this two `setState`s can be merged into one

File: src/App.js [hunk 2]
```diff
@@ -1,19 +1,161 @@
...
+      let sorted = array.sort((item, nextItem) => (item.age < nextItem.age) ? -1 : (item.age > nextItem.age) ? 1 : 0);
+      this.setState({search: sorted, item: sorted[0], checkSortAge: !this.state.checkSortAge}) 
+    }
+    else {
+      let sorted = array.sort((item, nextItem) => (item.age < nextItem.age) ? 1 : (item.age > nextItem.age) ? -1 : 0);
+      this.setState({search: sorted, item: sorted[0], checkSortAge: !this.state.checkSortAge})
+    }
+  }
+  handleClear() {
+    let array = this.state.searched.length == 0 ? this.state.data : !this.state.checkSearch ? this.state.searched : this.state.data;
```
Comment (KatSick) [inline src/App.js]: consider using `const` instead of `let`

File: src/App.js [hunk 3]
```diff
@@ -1,19 +1,161 @@
...
+      let sorted = array.sort((item, nextItem) => (item.name.trim() < nextItem.name.trim()) ? -1 : (item.name.trim() > nextItem.name.trim()) ? 1 : 0);
+      this.setState({search: sorted, item: sorted[0], checkSortName: !this.state.checkSortName}) 
+    }
+    else {
+      let sorted = array.sort((item, nextItem) => (item.name.trim() < nextItem.name.trim()) ? 1 : (item.name.trim() > nextItem.name.trim()) ? -1 : 0);
+      this.setState({search: sorted, item: sorted[0], checkSortName: !this.state.checkSortName})
+    }
+  }
+  handleAgeSort(){
+    let array = this.state.searched.length == 0 ? this.state.data : !this.state.checkSearch ? this.state.searched : this.state.data;
```
Comment (KatSick) [inline src/App.js]: looks like this checks is duplicated in several methods, what about DRY (Do not repeat yourself) practice ?

File: src/App.js [hunk 4]
```diff
@@ -1,19 +1,161 @@
...
+      .then((data) => {
+        this.setState({data: data, store: data, item: data[0]})
+      });
+  }
+  handleClick(item) {
+    this.setState({item: item})
+  }
+  handleSearch(){
+    let search = this.state.data.filter((item,index)=>item.name.includes(this.refs.search.value));
+    search.length == 0 ? this.setState({checkSearch: true, searched: [], item: {}}) :
```
Comment (KatSick) [inline src/App.js]: same regarding doubled `setState`s

File: src/App.js [hunk 5]
```diff
@@ -1,19 +1,161 @@
...
     // Example of data fetching from server
     fetch('data.json')
       .then((httpResponse) => httpResponse.json())
-      .then((data) => console.log(data));
+      .then((data) => {
+        this.setState({data: data, store: data, item: data[0]})
+      });
+  }
+  handleClick(item) {
+    this.setState({item: item})
```
Comment (KatSick) [inline src/App.js]: Try to use object literal property value shorthand here like `this.setState({ item });`

**Review by KatSick — CHANGES_REQUESTED** [review]
There is some changes required to pass test assignment.

**Comment by Maksymyr** [comment]
Hello, Ostap!
Sorry for late message, but I didn't expect your answer today))
Just recently saw it :(
I put it here:
https://maksymyr.github.io/strikersoft-frontend-test-assignment/

Thank you for allotted time and good night!

Kind regards,
Maksym

2017-11-27 21:37 GMT+02:00 Ostap Chervak <notifications@github.com>:

**Comment by KatSick** [comment]
@Maksymyr not a problem. checked your URL. looks OK.
however, i have some questions on your code. please check out review above.

**Comment by Maksymyr** [comment]
I saw, thank you for feedback!
I'll inspect and improve it. Trying to make it faster I made many extra

2017-11-28 2:23 GMT+02:00 Ostap Chervak <notifications@github.com>:

**Comment by KatSick** [comment]
@Maksymyr thanks for your motivation to make it better, however our goal is to finilize your codebase.
try to resolve my small comments and thats all.

**Comment by Maksymyr** [comment]
Be sure, I'll study it now, but I'll probably fix it tomorrow.
My gratitude for conveying of your experience!

2017-11-28 2:28 GMT+02:00 Ostap Chervak <notifications@github.com>:

**Review by KatSick — APPROVED** [review]

**Comment by KatSick** [comment]
@Maksymyr thanks for your feedback. looks ok for me

**Comment by Maksymyr** [comment]
Thank you for comments and attention, Ostap! 
Best regards!

**Pull request closed without merging**